packages/babel-core/src/config/config-chain.js

```diff
@@ -148,9 +144,19 @@ export function buildRootChain(
 
   const configFileChain = emptyChain();
   if (configFile) {
-    const result = loadFileChain(configFile, context);
+    const validatedFile = validateConfigFile(configFile);
+    const result = loadFileChain(validatedFile, context);
     if (!result) return null;
 
+    // Allow config files to toggle `.babelrc` resolution on and off and
+    // specify where the roots are.
+    if (babelrc === undefined) {
+      babelrc = validatedFile.options.babelrc;
+    }
+    if (babelrcRoots === undefined) {
```

Comment:
I'm always hesitant to do merging for anything that isn't key/value pairs, because it's not as clear what the user expects. I'd rather not encourage people to specify this in two places, anyway.

Reply:
Then we should at least warn instead of silently ignoring some values.

Reply:
Hmm, maybe. I worry that it would be annoying if the user was specifically _trying_ to override the value.